src/main/java/medicaldoctor/controllers/MyPatientController.java

```diff
@@ -0,0 +1,116 @@
+
+public class MyPatientController implements Initializable{
+    @FXML
+    private TableView<Patient> table;
+    
+    @FXML
+    private TableColumn<Patient, String> c_first_name, c_last_name, c_sex, c_ins;
+    
+    @FXML
+    private TextField first_name_tf, last_name_tf;
+    
+    @FXML
+    private TableColumn<Patient, Integer> c_age;
+    
+    @FXML
+    private List<Patient> patients;
+    
+    @FXML
+    private List<Patient> doctors_patients;
+    
+    void search(TextField tf, int i){
+        if(!tf.getText().isEmpty()){
+            if(patients.get(i).getFirstName().toLowerCase().equals(tf.getText().toLowerCase()))
+                getTableResults(patients.get(i));
+            if(patients.get(i).getLastName().toLowerCase().equals(tf.getText().toLowerCase()))
+                getTableResults(patients.get(i));
+            
+        }
+    }
+    
+    void getTableResults(Patient patients){
+        table.getItems().clear();
+        c_first_name.setCellValueFactory(new PropertyValueFactory("firstName"));
+        c_last_name.setCellValueFactory(new PropertyValueFactory("lastName"));
+        c_age.setCellValueFactory(new PropertyValueFactory("age"));
+        c_sex.setCellValueFactory(new PropertyValueFactory("sex"));
+        c_ins.setCellValueFactory(new PropertyValueFactory("medicalInsurance"));
+        ObservableList<Patient> data = FXCollections.<Patient>observableArrayList();
+        data.addAll(patients);
+        table.setItems(data);
+        table.refresh();
+    }
+    
+    @FXML
+    void search_patient(ActionEvent event){
+        for(int i =0; i < patients.size(); i++){
+            search(first_name_tf,i);
+            search(last_name_tf,i);
+        }
+    }
+    
+    @Override
+    public void initialize(URL location, ResourceBundle resources){
+        doctors_patients = new ArrayList<>();
+        
+        try (DatabaseScope scope = new DatabaseScope()) {
+            patients = Patient.getAll();
+        } catch (Exception ex) {
+           throw new InitializeException(ex);
+        }
+        
+        for(int i=0; i<patients.size();i++){
+            if(AppSession.getActiveUser().getFullName().toLowerCase().equals(patients.get(i).getPrimaryDoctor().getFullName().toLowerCase())){
+                doctors_patients.add(patients.get(i));
+            }
+        }
+        c_first_name.setCellValueFactory(new PropertyValueFactory("firstName"));
+        c_last_name.setCellValueFactory(new PropertyValueFactory("lastName"));
+        c_age.setCellValueFactory(new PropertyValueFactory("age"));
+        c_sex.setCellValueFactory(new PropertyValueFactory("sex"));
+        c_ins.setCellValueFactory(new PropertyValueFactory("medicalInsurance"));
+        ObservableList<Patient> data = FXCollections.<Patient>observableArrayList();
+        data.addAll(doctors_patients);
+        table.setItems(data);
+        table.refresh();
+        
+        table.setRowFactory(tv -> {
```

Comment:
you may want to practice better design in the first place. i see a bunch of copy and paste which means you are doing something wrong.
better design (at least) provides: 1. better maintenance since  you only have to change things once, 2. easier to read because modularity, and 3. quicker future development.

Reply:
No? how does that mean it's wrong? It allows you to click the rows on the table.

Reply:
not wrong necessarily in functionality, but wrong in design